### PR TITLE
required changes for twitter quick reply feature

### PR DESCRIPTION
**Pull request by abhishekchotaliya**
These changes are required for twitter dm package to support quick reply feature.

### CODE REVIEW

**Comment by jdwuarin** [comment]
This is not how botmaster is supposed to work. There should never be any mention of any potential. Underlying message format. All the formatting should happen in whichever class extends `BaseBot`. This project should not have to change to support it

File: lib/base_bot.js [hunk 1]
```diff
@@ -246,44 +246,49 @@ class BaseBot extends EventEmitter {
...
+        return this.__formatOutgoingMessage(outgoingMessage, sendOptions);
+      })
+      .then((rawMessage) => {
+        responseBody.sentRawMessage = rawMessage;
+        // twitter_qr is a flag to identify if message is for twitter platform
+        let twitter_qr = false
+        if (outgoingMessage.quick_replies) {
+          twitter_qr = true
+        }
+        return this.__sendMessage(rawMessage, sendOptions, twitter_qr);
```
Comment (jdwuarin) [inline lib/base_bot.js]: The `rawMessage` object should contain enough information for you to know that this is a quick_reply in the `twitter-dm` package

File: lib/base_bot.js [hunk 2]
```diff
@@ -548,14 +553,26 @@ class BaseBot extends EventEmitter {
...
+        quickReplies.push({
+          content_type: 'text',
+          title: buttonTitle,
+          payload: buttonTitle, // indeed, in default mode payload is buttonTitle
+        });
+      }
     }
-    outgoingMessage.addQuickReplies(quickReplies);
+    outgoingMessage.addQuickReplies(this.type, quickReplies);
     return this.sendMessage(outgoingMessage, arguments[3]);
```
Comment (jdwuarin) [inline lib/base_bot.js]: `this.type` is also available in the `botmaster-twitter-dm` BotClass object. No need to include it here. `addQuickReplies` should have a different behaviour, this should be done by overriding the method in the `TwitterDMBot` class. But it doesn't look necessary here.

File: lib/base_bot.js [hunk 3]
```diff
@@ -548,14 +553,26 @@ class BaseBot extends EventEmitter {
...
+      for (const buttonTitle of buttonTitles) {
+        quickReplies.push({
+          content_type: 'text',
+          title: buttonTitle,
+          payload: buttonTitle, // indeed, in default mode payload is buttonTitle
+        });
+      }
     }
-    outgoingMessage.addQuickReplies(quickReplies);
+    outgoingMessage.addQuickReplies(this.type, quickReplies);
```
Comment (jdwuarin) [inline lib/base_bot.js]: `this.type` is also available in the `botmaster-twitter-dm` BotClass object. No need to include it here. `addQuickReplies` should have a different behaviour, this should be done by overriding the method in the `TwitterDMBot` class. But it doesn't look necessary here.

**Comment by abhishekchotaliya** [comment]
Thanks for the suggestions!
Will implement the necessary changes very soon.

**Pull request closed without merging**